### PR TITLE
feat: add hooks for static RPC provider, refactor some utils functions

### PR DESCRIPTION
**Pull request by solidovic**
### Description

Added and use hooks:
- useMainnetStaticRpcProvider
- useCurrentStaticRpcProvider

Some utils functions not creates StaticRpcProvider because of pass provider as prop, see:
- stakeProcessing
- resolveEns
- stEthEthRequest
- getFeeHistory
- getFeeData
- etc

Refactoring an import order in changed files.

### Code review notes

There are many TODOs

### Checklist:

- [x] Checked the changes locally.
- [ ] Created / updated analytics events.
- [ ] Created / updated the technical documentation (README.md / [docs](https://docs.lido.fi/) / etc.).
- [ ] Affects / requires changes in other services (Matomo / Sentry / CloudFlare / etc.).

### CODE REVIEW

File: config/rpc.ts [hunk 1]
```diff
@@ -4,34 +4,40 @@ import { useSDK } from '@lido-sdk/react';
 import { useCustomConfig } from 'providers/custom-config';
 import { CHAINS } from 'utils/chains';
 
-export const getBackendRPCPath = (chainId: string | number): string => {
+import dynamics from './dynamics';
+
+// export const getRpcUrl = (): string => {
```
Comment (Jeday) [inline config/rpc.ts:9]: No commented code. Just delete. We have git history.

File: config/rpc.ts [hunk 2]
```diff
@@ -4,34 +4,40 @@ import { useSDK } from '@lido-sdk/react';
...
+
+export const getBackendRPCPath = (
+  chainId: string | number,
+): string | undefined => {
   const BASE_URL = typeof window === 'undefined' ? '' : window.location.origin;
   return `${BASE_URL}/api/rpc?chainId=${chainId}`;
 };
 
-export const useGetRpcUrl = () => {
+export const useRpcUrlByChainIdGetter = () => {
```
Comment (Jeday) [inline config/rpc.ts:21]: Why getter? That's just a hook.

Comment (Jeday) [reply config/rpc.ts:21]: For non react runtime, naming would be getRpcUrl...

File: features/ipfs/ipfs-info-box/ipfs-info-box.tsx [hunk 3]
```diff
@@ -32,7 +32,8 @@ export const IPFSInfoBox = () => {
   const { data: rpcCheckResult, initialLoading: isLoading } = useLidoSWR(
     `rpc-url-check-${rpcUrl}-${chainId}`,
     async () => {
-      return await checkRpcUrl(rpcUrl, chainId);
+      // TODO
```
Comment (Jeday) [inline features/ipfs/ipfs-info-box/ipfs-info-box.tsx:35]: specify TODO so that others can pick up later

File: providers/custom-config.tsx [hunk 4]
```diff
@@ -44,6 +42,14 @@ const DEFAULT_STATE: SavedCustomConfig = {
   rpcUrls: {},
 };
 
+// export const getRestoredSettings = () => {
```
Comment (Jeday) [inline providers/custom-config.tsx:45]: delete comments

File: providers/web3.tsx [hunk 5]
```diff
@@ -27,7 +44,8 @@ const Web3Provider: FC<PropsWithChildren> = ({ children }) => {
       provider: () =>
         getStaticRpcBatchProvider(
           chain.id,
-          getBackendRPCPath(chain.id),
+          // TODO
```
Comment (Jeday) [inline providers/web3.tsx:47]: what TODO?

File: providers/web3.tsx [hunk 6]
```diff
@@ -4,15 +4,32 @@ import { getConnectors, holesky } from 'reef-knot/core-react';
...
 const Web3Provider: FC<PropsWithChildren> = ({ children }) => {
   const { defaultChain, supportedChainIds, walletconnectProjectId } =
     useCustomConfig();
 
+  const getRpcUrlByChainId = useRpcUrlByChainIdGetter();
+
+  const backendRPC = useMemo(
+    () =>
+      supportedChainIds.reduce<Record<number, string>>(
+        // TODO
```
Comment (Jeday) [inline providers/web3.tsx:22]: what TODO

File: providers/web3.tsx [hunk 7]
```diff
@@ -4,15 +4,32 @@ import { getConnectors, holesky } from 'reef-knot/core-react';
...
+        (res, curr) => ({ ...res, [curr]: getRpcUrlByChainId(curr) ?? '' }),
+        {
+          // TODO
```
Comment (Jeday) [inline providers/web3.tsx:25]: what TODO

File: shared/hooks/use-current-static-rpc-provider.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+export const useCurrentStaticRpcProvider = (): {
+  staticRpcProvider: StaticJsonRpcBatchProvider;
+  chainId: CHAINS;
+} => {
+  const { chainId } = useSDK();
+  const rpcUrl = useRpcUrl();
+
+  return {
+    // TODO
```
Comment (Jeday) [inline shared/hooks/use-current-static-rpc-provider.ts:16]: what TODO

**Review by Jeday — CHANGES_REQUESTED** [review]

File: config/rpc.ts (hunk 1 above)
Comment (solidovic) [reply config/rpc.ts:9]: removed

File: config/rpc.ts (hunk 2 above)
Comment (solidovic) [reply config/rpc.ts:21]: logically, but 2 verbs step by step not beautiful and bad readable.
reverted and set TODO.

File: features/ipfs/ipfs-info-box/ipfs-info-box.tsx (hunk 3 above)
Comment (solidovic) [reply features/ipfs/ipfs-info-box/ipfs-info-box.tsx:35]: remove

File: providers/custom-config.tsx (hunk 4 above)
Comment (solidovic) [reply providers/custom-config.tsx:45]: remove

File: providers/web3.tsx (hunk 5 above)
Comment (solidovic) [reply providers/web3.tsx:47]: remove

File: providers/web3.tsx (hunk 6 above)
Comment (solidovic) [reply providers/web3.tsx:22]: remove

File: providers/web3.tsx (hunk 7 above)
Comment (solidovic) [reply providers/web3.tsx:25]: remove

File: shared/hooks/use-current-static-rpc-provider.ts (hunk 8 above)
Comment (solidovic) [reply shared/hooks/use-current-static-rpc-provider.ts:16]: remove

**Review by Jeday — APPROVED** [review]

File: config/rpc.ts [hunk 9]
```diff
@@ -1,37 +1,41 @@
...
+  const getRpcUrlByChainId = useGetRpcUrlByChainId();
   // TODO: use `satisfies` when will be TS v5
-  return getRpcUrl(chainId as unknown as CHAINS);
-}
-
-export const backendRPC = {
-  [CHAINS.Mainnet]: getBackendRPCPath(CHAINS.Mainnet),
-  [CHAINS.Goerli]: getBackendRPCPath(CHAINS.Goerli),
-  [CHAINS.Holesky]: getBackendRPCPath(CHAINS.Holesky),
+  return getRpcUrlByChainId(chainId as unknown as CHAINS);
```
Comment (DiRaiks) [inline config/rpc.ts:40]: `as number` is enough

File: features/home/stake-form/stake-form.tsx [hunk 10]
```diff
@@ -7,7 +7,9 @@ import {
   useEffect,
   useRef,
 } from 'react';
+import { useWeb3 } from 'reef-knot/web3-react';
 import { useRouter } from 'next/router';
+
```
Comment (DiRaiks) [inline features/home/stake-form/stake-form.tsx:12]: unnecessary empty string

File: features/home/stake-form/stake-form.tsx [hunk 11]
```diff
@@ -27,19 +28,22 @@ import {
 } from '@lidofinance/lido-ui';
 import { L2Swap } from 'shared/banners';
 import { DATA_UNAVAILABLE } from 'config';
+import { OneinchInfo } from 'features/home/oneinch-info/oneinch-info';
```
Comment (DiRaiks) [inline features/home/stake-form/stake-form.tsx:31]: Is not used in this file

File: features/rewards/components/stats/Stats.tsx [hunk 12]
```diff
@@ -31,16 +32,17 @@ export const Stats: FC<StatsProps> = (props) => {
...
     if (dynamics.defaultChain !== 1) {
       setStEthEth(constants.WeiPerEther);
     } else {
-      const stEthEth = await stEthEthRequest();
+      const stEthEth = await stEthEthRequest(mainnetStaticRpcProvider);
 
       setStEthEth(stEthEth);
     }
-  }, []);
+  }, [mainnetStaticRpcProvider]);
```
Comment (DiRaiks) [inline features/rewards/components/stats/Stats.tsx:45]: mainnetStaticRpcProvider is cached?

File: features/home/stake-form/stake-form.tsx [hunk 13]
```diff
@@ -109,6 +115,7 @@ export const StakeForm: FC = memo(() => {
       );
     },
     [
+      staticRpcProvider,
```
Comment (DiRaiks) [inline features/home/stake-form/stake-form.tsx:118]: staticRpcProvider is cached?

File: features/withdrawals/hooks/contract/useRequest.ts [hunk 14]
```diff
@@ -43,6 +45,7 @@ const useWithdrawalRequestMethods = () => {
       invariant(account, 'must have account');
       invariant(signature, 'must have signature');
       invariant(contractWeb3, 'must have contractWeb3');
+      invariant(providerWeb3, 'must have providerWeb3');
```
Comment (DiRaiks) [inline features/withdrawals/hooks/contract/useRequest.ts:48]: I don't see providerWeb3 being used in this function

File: features/withdrawals/hooks/contract/useRequest.ts [hunk 15]
```diff
@@ -99,6 +111,7 @@ const useWithdrawalRequestMethods = () => {
...
```
Comment (DiRaiks) [inline features/withdrawals/hooks/contract/useRequest.ts:114]: I don't see providerWeb3 being used in this function

File: features/home/stake-form/stake-form.tsx (hunk 10 above)
Comment (solidovic) [reply features/home/stake-form/stake-form.tsx:12]: many imports. wants grouped

File: features/rewards/components/stats/Stats.tsx (hunk 12 above)
Comment (solidovic) [reply features/rewards/components/stats/Stats.tsx:45]: added useMemo to useCurrentStaticRpcProvider and useMainnetStaticRpcProvider

File: features/home/stake-form/stake-form.tsx (hunk 13 above)
Comment (solidovic) [reply features/home/stake-form/stake-form.tsx:117]: added useMemo to useCurrentStaticRpcProvider and useMainnetStaticRpcProvider

File: features/withdrawals/hooks/contract/useRequest.ts (hunk 15 above)
Comment (solidovic) [reply features/withdrawals/hooks/contract/useRequest.ts]: removed

File: features/withdrawals/hooks/contract/useRequest.ts (hunk 14 above)
Comment (solidovic) [reply features/withdrawals/hooks/contract/useRequest.ts]: removed

File: features/home/stake-form/stake-form.tsx (hunk 11 above)
Comment (solidovic) [reply features/home/stake-form/stake-form.tsx]: removed

File: config/rpc.ts (hunk 9 above)
Comment (solidovic) [reply config/rpc.ts]: fixed

**Review by DiRaiks — APPROVED** [review]

**Pull request merged**